packages/jest-core/src/getConfigsOfProjectsToRun.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {Config} from '@jest/types';
+import getProjectDisplayName from './getProjectDisplayName';
+
+export default function getConfigsOfProjectsToRun(
+  argv: Config.Argv,
+  projectConfigs: Array<Config.ProjectConfig>,
+): Array<Config.ProjectConfig> {
+  if (!argv.runProjects) {
+    return projectConfigs;
+  }
+  const namesOfProjectsToRun = new Set<string>(argv.runProjects);
+  return projectConfigs.filter(config => {
+    const name = getProjectDisplayName(config);
+    return name && namesOfProjectsToRun.has(name);
```

Comment:
Should we print a warning if `name` is not defined?

Reply:
It is not. Doing it right now.

Reply:
This is now addressed here: https://github.com/yacinehmito/jest/blob/1c35afb7a9e7265d9547f031b8ec6e1272824390/packages/jest-core/src/getSelectProjectsMessage.ts#L33

We can discuss the wording of the warning.